src/tests/test_remote.py

```diff
@@ -14,68 +15,110 @@ class TestMyGenesetDataIntegrity(MyGenesetWebTestBase):
-        assert kegg['hits'][0]['source'] == 'kegg'
+        kegg = self.query(q="source:kegg", fields="all")
+        assert "kegg" in kegg["hits"][0].keys()
+        assert kegg["hits"][0]["source"] == "kegg"
 
     def test_query_by_source_do(self):
-        do = self.query(q='source:do', fields='all')
-        assert 'do' in do['hits'][0].keys()
-        assert do['hits'][0]['source'] == 'do'
+        do = self.query(q="source:do", fields="all")
+        assert "do" in do["hits"][0].keys()
+        assert do["hits"][0]["source"] == "do"
 
     def test_query_by_source_reactome(self):
-        reactome = self.query(q='source:reactome', fields='all')
-        assert 'reactome' in reactome['hits'][0].keys()
-        assert reactome['hits'][0]['source'] == 'reactome'
+        reactome = self.query(q="source:reactome", fields="all")
+        assert "reactome" in reactome["hits"][0].keys()
+        assert reactome["hits"][0]["source"] == "reactome"
 
     def test_query_by_source_smpdb(self):
-        smpdb = self.query(q='source:smpdb', fields='all')
-        assert 'smpdb' in smpdb['hits'][0].keys()
-        assert smpdb['hits'][0]['source'] == 'smpdb'
+        smpdb = self.query(q="source:smpdb", fields="all")
+        assert "smpdb" in smpdb["hits"][0].keys()
+        assert smpdb["hits"][0]["source"] == "smpdb"
+
+    def test_include_filter_check(self):
+        query_all = self.query(include="all")
+        query_curated = self.query(include="curated")
+        query_public = self.query(include="public")
+        query_user = self.query(include="user")
+        query_anonymous = self.query(include="anonymous")
+        assert query_all["total"] == query_curated["total"] + query_user["total"]
+        assert query_all["total"] >= query_public["total"]
+        assert query_user["total"] >= query_anonymous["total"]
+
+    def test_include_all_filter_plus_query(self):
+        query = self.query(q="glucose", species="9615", include="all")
+        assert query["hits"][0]["taxid"] == "9615"
+
+    def test_include_curated_filter_plus_query(self):
+        query = self.query(q="glucose", species="9615", include="curated")
+        assert query["hits"][0]["taxid"] == "9615"
+
+    def test_include_public_filter_plus_query(self):
+        query = self.query(q="glucose", species="9615", include="public")
+        assert query["hits"][0]["taxid"] == "9615"
+
+    def test_include_user_filter_plus_query(self):
+        res = self.request("query?q=glucose&species=9615&include=user").json()
+        assert res["total"] == 0
+        assert len(res["hits"]) == 0
+
+    def test_include_anonymous_filter_plus_query(self):
+        res = self.request("query?q=glucose&species=9615&include=anonymous").json()
+        assert res["total"] == 0
+        assert len(res["hits"]) == 0
+
+    def test_include_filter_error(self):
+        res = self.request("query?include=a_wrong_include", expect=400).json()
+        assert res == {
```

Comment:
WDYT about using map/dict in the assert?

Reply:
looks fine to me. If later you find any field could change, you can skip that field.